app/src/main/java/com/douglasbruce/tasky/core/data/repository/AgendaRepositoryImpl.kt

```diff
@@ -0,0 +1,29 @@
+package com.douglasbruce.tasky.core.data.repository
+
+import com.douglasbruce.tasky.core.domain.mapper.toAgendaItems
+import com.douglasbruce.tasky.core.domain.repository.AgendaRepository
+import com.douglasbruce.tasky.core.model.AgendaItem
+import com.douglasbruce.tasky.core.network.retrofit.RetrofitTaskyNetwork
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import java.time.LocalDate
+import java.time.LocalTime
+import java.time.ZonedDateTime
+import javax.inject.Inject
+
+class AgendaRepositoryImpl @Inject constructor(
+    private val taskyNetwork: RetrofitTaskyNetwork,
+) : AgendaRepository {
+    override suspend fun getAgendaForDate(date: LocalDate): Flow<List<AgendaItem>> {
+        //TODO: Read locally first
+        return flow {
+            val zone = ZonedDateTime.now().zone
+            val agendaItems = taskyNetwork.getAgenda(
+                timeZone = zone.toString(),
+                time = ZonedDateTime.of(date, LocalTime.now(), zone).toEpochSecond() * 1000
+            ).toAgendaItems()
+
+            emit(agendaItems)
+        }
+    }
```

Comment:
This is one way to do it, but over time I found that the simpler way is to just read the stream of data from the DB and return it right away. You can then have an additional function which can be triggered to refresh the cache. As soon as new items are inserted, the DB will automatically emit them anyways. Something like this:
```
override suspend fun getAgendaForDate(date: LocalDate): Flow<List<AgendaItem>> {
    return combine(
        eventDao.getEventsForDate(date),
        taskDao.getTasksForDate(date),
        reminderDao.getRemindersForDate(date),
    ) { eventEntities, taskEntities, reminderEntities -> 
        val events = eventEntities.map { it.toEvent() }
        val tasks = taskEntities.map { it.toTask() }
        val reminders = reminderEntities.map { it.toReminder() }
        (events + tasks + reminders).sortedBy { it.startTime }
    }
}

override suspend fun fetchAgendaForDate(date: LocalDate) {
    // Call to /agenda route
    // Insert new items in DB, which will trigger above's flow
}
```
This way it's easier to trigger a refresh than with keeping everything in one flow, IMO.